rocketpy/simulation/flight.py

```diff
@@ -2639,35 +2615,105 @@ def drag_power(self):
+        stream_vz_body = (
+            -self.attitude_vector_x.y_array * self.stream_velocity_x.y_array
+            - self.attitude_vector_y.y_array * self.stream_velocity_y.y_array
+            - self.attitude_vector_z.y_array * self.stream_velocity_z.y_array
+        )
         # Define freestream speed list
-        free_stream_speed = [self.free_stream_speed.get_value_opt(i) for i in self.time]
-        free_stream_speed = np.nan_to_num(free_stream_speed)
+        free_stream_speed = self.free_stream_speed.y_array
 
-        # Normalize dot product
-        dot_product_normalized = [
-            i / j if j > 1e-6 else 0 for i, j in zip(dot_product, free_stream_speed)
-        ]
-        dot_product_normalized = np.nan_to_num(dot_product_normalized)
-        dot_product_normalized = np.clip(dot_product_normalized, -1, 1)
+        stream_vz_body_normalized = np.divide(
+            stream_vz_body,
+            free_stream_speed,
+            out=np.zeros_like(stream_vz_body),
+            where=free_stream_speed > 1e-6,
+        )
+        stream_vz_body_normalized = np.clip(stream_vz_body_normalized, -1, 1)
 
         # Calculate angle of attack and convert to degrees
-        angle_of_attack = np.rad2deg(np.arccos(dot_product_normalized))
+        angle_of_attack = np.rad2deg(np.arccos(stream_vz_body_normalized))
+        angle_of_attack = np.nan_to_num(angle_of_attack)
 
         return np.column_stack([self.time, angle_of_attack])
 
+    @funcify_method("Time (s)", "Partial Angle of Attack (°)", "spline", "constant")
+    def partial_angle_of_attack(self):
+        """Partial angle of attack of the rocket with respect to the stream
+        velocity vector. By partial angle of attack, it is meant the angle
+        between the stream velocity vector in the y-z plane and the rocket's
+        z-axis. All in the Body Axes Coordinate System."""
+        # Stream velocity in standard aerodynamic frame
+        stream_velocity = -self.stream_velocity_body_frame
+        alpha = np.arctan2(
+            stream_velocity[:, 1],
+            stream_velocity[:, 2],
+        )  # y-z plane
+        return np.column_stack([self.time, np.rad2deg(alpha)])
+
+    @funcify_method("Time (s)", "Beta (°)", "spline", "constant")
+    def angle_of_sideslip(self):
+        """Angle of sideslip of the rocket with respect to the stream
+        velocity vector. Defined as the angle between the stream velocity
+        vector in the x-z plane and the rocket's z-axis. All in the Body
+        Axes Coordinate System."""
+        # Stream velocity in standard aerodynamic frame
+        stream_velocity = -self.stream_velocity_body_frame
+        beta = np.arctan2(
+            -stream_velocity[:, 0],
+            stream_velocity[:, 2],
+        )  # x-z plane
+        return np.column_stack([self.time, np.rad2deg(beta)])
+
+    @funcify_method("Time (s)", "Angle of Attack (°)", "spline", "constant")
+    def angle_of_attack2(self):
```

Comment:
what is the difference between the `angle_of_attack` and the `angle_of_attack2`?